File: go/pkg/step/start.go
package step

import (
	"bufio"
	"fmt"
	"strconv"
	"strings"

	"github.com/argenkiwi/ambler/go/pkg/lead"
)

func Start(state int, reader *bufio.Reader) (int, lead.Lead, error) {
	fmt.Print("Enter a starting number (or press Enter for default 0): ")
	input, err := reader.ReadString('\n')
	if err != nil {
		return state, lead.Start, err
	}
	input = strings.TrimSpace(input)

	if input == "" {
		fmt.Printf("Starting count from default: %d\n", state)
		return state, lead.Count, nil
	}

	number, err := strconv.Atoi(input)
	if err != nil {
		fmt.Println("Invalid input. Please, try again.")
		return state, lead.Start, nil
	}

	fmt.Printf("Starting count from: %d\n", number)
	return number, lead.Count, nil
}

File: javascript/src/step/start.js
import { Lead } from "../lead.js";

export function start(state, readlineInterface) {
    return new Promise((resolve) => {
        readlineInterface.question("Enter a starting number (or press Enter for default 0): ", (input) => {
            if (input === "") {
                console.log(`Starting count from default: ${state}`);
                resolve([state, Lead.Count]);
            } else {
                const number = parseInt(input, 10);
                if (!isNaN(number)) {
                    console.log(`Starting count from: ${number}`);
                    resolve([number, Lead.Count]);
                } else {
                    console.log("Invalid input. Please, try again.");
                    resolve([state, Lead.Start]);
                }
            }
        });
    });
}